secure/key-manager/src/main.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Usage: ./key-manager node.config
+
+#![forbid(unsafe_code)]
+
+use libra_config::config::{KeyManagerConfig, NetworkConfig, NodeConfig, SecureBackend};
+use libra_key_manager::{libra_interface::JsonRpcLibraInterface, Error, KeyManager};
+use libra_secure_json_rpc::JsonRpcClient;
+use libra_secure_storage::{InMemoryStorage, OnDiskStorage, VaultStorage};
+use libra_secure_time::RealTimeService;
+use std::{env, process};
+
+// TODO(joshlind): initialize the metrics components for the key manager!
+fn main() {
+    let args: Vec<String> = env::args().collect();
+
+    if args.len() != 2 {
+        eprintln!("Incorrect number of parameters, expected a path to a config file");
+        process::exit(1);
+    }
+
+    let config = NodeConfig::load(&args[1]).unwrap_or_else(|e| {
+        eprintln!("Unable to read provided config: {}", e);
+        process::exit(1);
+    });
+
+    if config.validator_network.is_none() {
+        eprintln!("Validator config missing from node config");
+        process::exit(1);
+    }
+    let network_config = config.validator_network.unwrap();
+    let key_manager_config = config.secure.key_manager;
+
+    create_and_execute_key_manager(network_config, key_manager_config).unwrap_or_else(|e| {
+        eprintln!("The Key Manager has failed during execution: {}", e);
+        process::exit(1);
+    });
+}
+
+fn create_and_execute_key_manager(
```

Comment:
This feels like something that should be in a testable library, but I think the type safety should be okay for now. Also do we really want a key manager talking to disk or in memory? I think we do not and shouldn't even bother implementing that path.

Reply:
1. I mean this is mostly plumbing code, right? I guess we could add a function to the lib, but like you say, type safety should be enough. This may change with our automated tests though!
2. Regarding supporting in_memory and on_disk, I see what you're saying. However, it did make testing this change manually much much easier. I think things would have been a lot more complicated if we forced manually testing to be done using vault? Thoughts?

Reply:
As discussed offline (for 2), we'll just remove support for on_disk and in_memory as automated tests will just use vault.